api/contract.go

```diff
@@ -44,6 +45,59 @@ type createFilterResponse struct {
 	model.Filter
 }
 
+// createFilterRequest is the request body for POST /filters
+type createFilterOutputsRequest struct {
+	State     string             `bson:"state" json:"state"`
+	Downloads model.FilterOutput `bson:"downloads"      json:"downloads"`
+}
+
+func isNotFullyPopulated(fi *model.FileInfo) error {
```

Comment:
This function would be better as a method on the *model.FileInfo type, as it has quite a generic name but takes a specific type